ext/strscan/strscan.c

```diff
@@ -1262,9 +1263,12 @@ strscan_captures(VALUE self)
     new_ary  = rb_ary_new2(num_regs);
 
     for (i = 1; i < num_regs; i++) {
-        VALUE str = extract_range(p,
-                                  adjust_register_position(p, p->regs.beg[i]),
-                                  adjust_register_position(p, p->regs.end[i]));
+        if (p->regs.beg[i] == -1)
```

Comment:
Could you declare this variable in this scope to narrow scope as much as possible?

```suggestion
        VALUE str;
        if (p->regs.beg[i] == -1)
```

Reply:
OK.
I See.